shopinvader_api_sale_loyalty/schemas/cart.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tools.safe_eval import safe_eval
+
+from odoo.addons.shopinvader_api_cart.schemas import CartResponse as BaseCartResponse
+
+from ..schemas import LoyaltyCard, LoyaltyProgram
+
+
+class CartResponse(BaseCartResponse, extends=True):
+    promo_codes: list[str] = []
+    # TODO: in v1, we had the applied coupons in the model.
+    # Not sure it is useful as we have the coupon codes in promo_codes...
+    reward_amount: float = 0
+    programs: list[LoyaltyProgram] = []
+    generated_coupons: list[LoyaltyCard] = []
```

Comment:
1. Should we add programs related info directly to the cart response?
For now, the problem is that as we extend the CartResponse Pydantic model, we should give read access to shopinvader_cart_user_group to all loyalty models.

2. I've added fields that seem useful. I don't think adding applied coupons is useful (as we have the related promo codes and the details on the sale order lines).